cdc/scheduler/internal/v3/coordinator.go

```diff
@@ -260,10 +263,19 @@ func (c *coordinator) poll(
 	}
 	msgBuf = append(msgBuf, msgs...)
 
+	pdTime := time.Now()
+	// only nil in unit test
+	if c.upstream != nil {
+		pdTime, err = c.upstream.PDClock.CurrentTime()
```

Comment:
it looks we only need a `PDClock`, so just pass a `PDClock` instead of the `upstream` to the coordinator should be enough.